crates/rpc/rpc-eth-api/src/core.rs

```diff
@@ -369,7 +369,7 @@ impl<T>
         RpcReceipt<T::NetworkTypes>,
     > for T
 where
-    T: FullEthApi,
+    T: FullEthApi<NetworkTypes: Network<ReceiptResponse = AnyTransactionReceipt>>,
```

Comment:
this is the opposite of what we want, we want to remove this constraint. however, the possibility of doing so depends partly on https://github.com/alloy-rs/alloy/issues/1245.